src/colorModels/oklab.ts

```diff
@@ -0,0 +1,68 @@
+import { RgbaColor, LabaColor, InputObject } from "../types";
+import { clamp, isPresent, round } from "../helpers";
+import { clampRgba, linearizeRgbChannel, unlinearizeRgbChannel } from "./rgb";
+import { ALPHA_PRECISION } from "../constants";
+
+// https://bottosson.github.io/posts/oklab/
+// https://ajalt.github.io/colormath/api/colormath/com.github.ajalt.colormath.model/-oklab/index.html
+
+export const clampOklaba = (oklaba: LabaColor): LabaColor => ({
+  l: clamp(oklaba.l, 0, 1),
+  a: clamp(oklaba.a, -1, 1),
+  b: clamp(oklaba.b, -1, 1),
+  alpha: clamp(oklaba.alpha),
+});
+
+export const roundOklaba = (oklaba: LabaColor): LabaColor => ({
+  l: round(oklaba.l, 4),
+  a: round(oklaba.a, 4),
+  b: round(oklaba.b, 4),
+  alpha: round(oklaba.alpha, ALPHA_PRECISION),
+});
+
+export const parseOklaba = ({ l, a, b, ok, alpha = 1 }: InputObject): RgbaColor | null => {
+  if (!isPresent(l) || !isPresent(a) || !isPresent(b) || !ok) return null;
+
+  const oklaba = clampOklaba({
+    l: Number(l),
+    a: Number(a),
+    b: Number(b),
+    alpha: Number(alpha),
+  });
+
+  return oklabaToRgba(oklaba);
+};
+
+export const rgbaToOklaba = (rgba: RgbaColor): LabaColor => {
```

Comment:
Should we add another type for `Oklab`?

Reply:
Yes, I can do that, but basically, they are the same. Why do you think we should add it?